subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JavaCompilerArgumentsBuilder.java

```diff
@@ -255,8 +255,21 @@ private void addClasspath() {
         }
 
         List<File> classpath = spec.getCompileClasspath();
-        args.add("-classpath");
-        args.add(classpath == null ? "" : Joiner.on(File.pathSeparatorChar).join(classpath));
+        List<File> modulePath = spec.getModulePath();
+        String moduleVersion = spec.getModuleVersion();
+
+        if (!classpath.isEmpty()) {
+            args.add("-classpath");
+            args.add(Joiner.on(File.pathSeparatorChar).join(classpath));
+        }
+        if (!modulePath.isEmpty()) {
```

Comment:
Can this have an effect on the pathing jar fix for too long commands? Is there an equivalent fix for the module path?

Reply:
This is not applicable here. We call/create the compiler via API `javax.tools.JavaComipler...` and there is no issue with a long argument list there.

The pathing JAR is a topic for execution and will be considered in #12428.